cdi/embedded/src/test/java/org/infinispan/cdi/embedded/test/cachemanager/programmatic/Config.java

```diff
@@ -18,6 +19,29 @@
  * @author Kevin Pollet <kevin.pollet@serli.com> (C) 2011 SERLI
  */
 public class Config {
+
+   /**
+    * <p>Associates the "large" cache with qualifier "large"</p>
+    *
+    * Note that {@link Named} works as a string-based qualifier, so it is necessary to support also {@link ConfigureCache}.
+    *
+    * @see Named
+    * @see ConfigureCache
+    */
+   @Named("large")
+   @ConfigureCache("large")
+   @Produces
+   public Configuration largeConfiguration = new ConfigurationBuilder().memory().size(10).build();
+
+   /**
+    * The same as the above. The intention here is to check whether we can use 2 Configurations with <code>@Named</code>
+    * annotations.
+    */
+   @Named("super-large")
+   @ConfigureCache("super-large")
+   @Produces
+   public Configuration superLargeConfiguration = new ConfigurationBuilder().memory().size(20).build();
```

Comment:
The exception only exists in 8.x. You're right. It's working correctly with 9.x.